test/e2e/storage/csi_volumes.go

```diff
@@ -245,16 +249,156 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
 			})
 		}
 	})
+
+	Context("CSI workload information [Feature:CSIDriverRegistry]", func() {
+		var (
+			driver         csiTestDriver
+			podInfoV1      = "v1"
+			podInfoUnknown = "unknown"
+			podInfoEmpty   = ""
+		)
+		BeforeEach(func() {
+			driver = initCSIMock(f, config)
+			driver.createCSIDriver()
+		})
+
+		AfterEach(func() {
+			driver.cleanupCSIDriver()
+		})
+
+		tests := []struct {
+			name                  string
+			podInfoOnMountVersion *string
+			driverExists          bool
+			expectPodInfo         bool
+		}{
+			{
+				name:                  "should not be passed when podInfoOnMountVersion=nil",
+				podInfoOnMountVersion: nil,
+				driverExists:          true,
+				expectPodInfo:         false,
+			},
+			{
+				name:                  "should be passed when podInfoOnMountVersion=v1",
+				podInfoOnMountVersion: &podInfoV1,
+				driverExists:          true,
+				expectPodInfo:         true,
+			},
+			{
+				name:                  "should not be passed when podInfoOnMountVersion=<empty string>",
+				podInfoOnMountVersion: &podInfoEmpty,
+				driverExists:          true,
+				expectPodInfo:         false,
+			},
+			{
+				name:                  "should not be passed when podInfoOnMountVersion=<unknown string>",
+				podInfoOnMountVersion: &podInfoUnknown,
+				driverExists:          true,
+				expectPodInfo:         false,
+			},
+			{
+				name:          "should not be passed when CSIDriver does not exist",
+				driverExists:  false,
+				expectPodInfo: false,
+			},
+		}
+
+		for _, t := range tests {
+			test := t
+			It(test.name, func() {
+				if test.driverExists {
+					driver := createCSIDriver(csics, "csi-mock-"+f.UniqueName, true, test.podInfoOnMountVersion)
```

Comment:
Shouldn't the controller driver-registrar sidecar container be the one creating the CSIDriver object?

Reply:
I think you're right; I removed this and the tests cases didn't fail (except `should be passed when podInfoOnMountVersion=v1` that fails anyway).

Reply:
Yes, but I think you will have to pass the podInfoOnMountVersion to the csi driver driver-registrar arguments:
https://github.com/kubernetes-csi/driver-registrar/blob/master/cmd/driver-registrar/main.go#L67

Reply:
Seems like I was wrong. This same approach is used in the `CSISkipAttach` test.

The registrar creates the CRD based on the manifest. Instead of creating the driver through the registrar, this test calls the function `createCSIDriver` because it needs to indicate which test cases the driver is available on (as well as the value of `podInfoOnMountVersion`). Note that `createCSIDriver` returns the CRD returned by the `clientset`, just like the registrar.

Another approach would be to have a different manifest file for each test case, but IMO that doesn't sound better.

Do you think we can keep it this way?

Reply:
Eventually I think we should test using the driver-registrar sidecar options since that is the official path.  This method is fine for now.